concepts/05 UI Components/DataGrid/10 Enhance Performance on Large Datasets/030 Rendering Optimization.md

```diff
@@ -1,7 +1,16 @@
+* **Render elements with complex content last**           
+Cells with complex content (filter row, command columns, editors and columns with checkboxes) can be rendered after cells with simple content. To do that, enable the [renderAsync](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#renderAsync') property. In addition, you can enable [the same property](/api-reference/_hidden/GridBaseColumn/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#renderAsync') for individual columns if they use a content-heavy [cellTemplate](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate').
 
-Cells with complex content (predefined templates, command columns, editors) can be rendered after cells with simple content. To do that, enable the [renderAsync](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#renderAsync') property. In addition, you can enable [the same property](/api-reference/_hidden/GridBaseColumn/renderAsync.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#renderAsync') for individual columns if they use a content-heavy [cellTemplate](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate').
+* **Prevent columns from adjusting their widths to the content**                    
+Keep the [columnAutoWidth](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#columnAutoWidth) property disabled. To optimize rendering even further, set fixed width for all columns. Specify it in the column's [width](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#width) property or use the [columnWidth](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#columnWidth) property to do this for all columns. Note that command columns ignore **columnWidth**, and you have to use the individual setting.
+
+* **Do not use checkboxes in Boolean columns**          
+Columns with *"boolean"* [dataType](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataType) use checkboxes to display column values. To disable this feature, assign **false** to the columns' [showEditorAlways](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#showEditorAlways) property. The checkboxes will be replaced with the **true** and **false** text values. You can change them in the [trueText](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#trueText) and [falseText](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#falseText) properties.
```

Comment:
Boolean should always be capitalized - it comes from somebody's last name. 

Again - there's no sentence that explicitly states that check boxes require more rendering resources than text. You give no clear reason why developers should disable them.

You also seem to be avoiding the phrase "by default", which is per guidelines... but you simply remove it from the sentence, and thus change the meaning of the sentence. You can still use the word "default", only avoid comma-separated clauses in the beginning of your sentences. 

```suggestion
Check boxes are the default display style for columns with *"Boolean"* [dataType](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataType). You can increase rendering speed if you instruct the grid to render text instead. To do this, assign **false** to the column's [showEditorAlways](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#showEditorAlways) property. Use the [trueText](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#trueText) and [falseText](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#falseText) properties to specify display strings. The default strings are **true** and **false**. 
```